lib/modules/dosomething/dosomething_reportback/dosomething_reportback.forms.inc

```diff
@@ -324,6 +324,11 @@ function dosomething_reportback_form_validate($form, &$form_state) {
 function dosomething_reportback_form_submit($form, &$form_state) {
   $values = $form_state['values'];
 
+  $campaign = node_load($values['nid']);
+  global $user;
+  $language_code = dosomething_global_get_language($user, $campaign);
+  $values['run_nid'] = $campaign->field_current_run[$language_code][0]['target_id'];
```

Comment:
I'm getting the campaigns current NID.

now that I've rethought about it, I think the procedure is this

campaign blah blah
source lang -> campaign run

campaign run
 en --> end date, title, etc
 global --> etc etc
 etc